### ISSUE TITLE
Type error on Mac dotnet 3.1.414 - expected Nullable<bool> in  Z1R_Avalonia/UI.fs

### ISSUE THREAD

**Issue by reznet** [brianmcn/Zelda1RandoTools#11]
When building on Mac platform with .net core SDK 3.1.414, UI.fs has a type error:

```
/Users/jeff/repos/Zelda1RandoTools/Z1R_Tracker/Z1R_Avalonia/UI.fs(48,92): error FS0001: This expression was expected to have type    'Nullable<bool>'    but here has type    'bool' [/Users/jeff/repos/Zelda1RandoTools/Z1R_Tracker/Z1R_Avalonia/Z1R_Avalonia.fsproj]
```

I was able to workaround it by declaring and using a Nullable<bool> but I don't know what the correct fix is, or if this is a bug in the Mac dotnet tools.

```
let isChecked = true
let nullableIsChecked = Nullable<bool> isChecked
let owGettableScreensCheckBox = new CheckBox(Content = owGettableScreensTextBox, IsChecked=nullableIsChecked)
```

**Comment by abraxas86** [comment]
Yeah, that Avalonia build is big broken... I was hoping to build it since the Windows version struggles in Wine on my Linux computer.

I think the issue is that it's targeting a really old version of the framework.

I tried to fix it up myself, but eventually threw in the towel and had Claude fix it...  I'm not proud of that, but I can say that it was able to make updates to at least get it to compile, though it seems to be missing a lot of features - I'm guessing this is an older build?

[EDIT] It still segfaults a lot, I think it probably needs a lot of work done on it :(

![Main Menu](https://raw.githubusercontent.com/abraxas86/Zelda1RandoTools/refs/heads/fix-avalonia-linux-build/doc/screenshots/avalonia-linux1.png)


![Tracker View](https://raw.githubusercontent.com/abraxas86/Zelda1RandoTools/refs/heads/fix-avalonia-linux-build/doc/screenshots/avalonia-linux2.png)


![Menu Square Marker](https://raw.githubusercontent.com/abraxas86/Zelda1RandoTools/refs/heads/fix-avalonia-linux-build/doc/screenshots/avalonia-linux3.png)
